python/GafferUI/ColorChooser.py

```diff
@@ -157,6 +163,297 @@ def _displayTransformChanged( self ) :
 		GafferUI.Slider._displayTransformChanged( self )
 		self._qtWidget().update()
 
+class _ColorField( GafferUI.Widget ) :
+
+	def __init__( self, color = imath.Color3f( 1.0 ), staticComponent = "h", **kw ) :
+
+		GafferUI.Widget.__init__( self, QtWidgets.QWidget(), **kw )
+
+		# \todo Allow the widget to grow if the containing window is resized. It should also
+		# be constrained to be square.
+		self._qtWidget().setMinimumSize( 216, 216 )
+		self._qtWidget().setSizePolicy( QtWidgets.QSizePolicy.Fixed, QtWidgets.QSizePolicy.Fixed )
+
+		self._qtWidget().paintEvent = Gaffer.WeakMethod( self.__paintEvent )
+
+		self.buttonPressSignal().connect( Gaffer.WeakMethod( self.__buttonPress ), scoped = False )
+		self.dragBeginSignal().connect( Gaffer.WeakMethod( self.__dragBegin ), scoped = False )
+		self.dragEnterSignal().connect( Gaffer.WeakMethod( self.__dragEnter ), scoped = False )
+		self.dragMoveSignal().connect( Gaffer.WeakMethod( self.__dragMove ), scoped = False )
+		self.dragEndSignal().connect( Gaffer.WeakMethod( self.__dragEnd ), scoped = False )
+
+		self.__valueChangedSignal = Gaffer.Signals.Signal2()
+
+		self.__color = color
+		self.__staticComponent = staticComponent
+		self.__colorFieldToDraw = None
+		self.setColor( color, staticComponent )
+
+	# Sets the color and the static component. `color` is in
+	# RGB space for RGB static components, HSV space for
+	# HSV static components and TMI space for TMI components.
+	def setColor( self, color, staticComponent ) :
+
+		self.__setColorInternal( color, staticComponent, GafferUI.Slider.ValueChangedReason.SetValues )
+
+	# Returns a tuple of the color and static component.
+	def getColor( self ) :
+
+		return self.__color, self.__staticComponent
+
+	## A signal emitted whenever a value has been changed. Slots should
+	# have the signature slot( _ColorField, GafferUI.Slider.ValueChangedReason )
+	def valueChangedSignal( self ) :
+
+		return self.__valueChangedSignal
+
+	## Returns True if a user would expect the specified sequence
+	# of changes to be merged into one undoable event.
+	@classmethod
+	def changesShouldBeMerged( cls, firstReason, secondReason ) :
+
+		if type( firstReason ) != type( secondReason ) :
+			return False
+
+		return ( firstReason, secondReason ) in (
+			# click and drag
+			( GafferUI.Slider.ValueChangedReason.Click, GafferUI.Slider.ValueChangedReason.DragBegin ),
+			( GafferUI.Slider.ValueChangedReason.DragBegin, GafferUI.Slider.ValueChangedReason.DragMove ),
+			( GafferUI.Slider.ValueChangedReason.DragMove, GafferUI.Slider.ValueChangedReason.DragMove ),
+			( GafferUI.Slider.ValueChangedReason.DragMove, GafferUI.Slider.ValueChangedReason.DragEnd ),
+		)
```

Comment:
I don't think this is called any more, and can be removed.

Reply:
Right! Fixed in https://github.com/GafferHQ/gaffer/pull/5944/commits/d9743061ffa181fbb428c9c904c3746660299926